src/fetchhead.c

```diff
@@ -118,7 +118,7 @@ int git_fetchhead_write(git_repository *repo, git_vector *fetchhead_refs)
 	if (git_buf_joinpath(&path, repo->gitdir, GIT_FETCH_HEAD_FILE) < 0)
 		return -1;
 
-	if (git_filebuf_open(&file, path.ptr, GIT_FILEBUF_FORCE, GIT_REFS_FILE_MODE) < 0) {
+	if (git_filebuf_open(&file, path.ptr, GIT_FILEBUF_APPEND, GIT_REFS_FILE_MODE) < 0) {
```

Comment:
Without having a look at the surrounding code it feels a bit strange that we reopen the buffer several times to append to it. But this is not a fault of the current PR, so I guess it's fine as-is.

Reply:
It doesn't feel like the cleanest way, but it does let us behave like git (which also reopens the file) without changing a whole lot of functions to propagate a multi-step creation of the file contents.

Reply:
Agreed - this seems a bit odd, but it is intentionally so to be compatible with git's behavior.